source/Calamari/Deployment/PackageRetention/Model/Journal.cs

```diff
@@ -123,9 +123,28 @@ public int GetUsageCount(PackageIdentity package)
             return GetUsage(package).Count();
         }
 
-        public void ExpireStaleLocks()
+        public void ExpireStaleLocks(TimeSpan timeBeforeExpiration)
         {
-            throw new NotImplementedException();
+            try
+            {
+                using (var repository = repositoryFactory.CreateJournalRepository())
+                {
+                    foreach (var entry in repository.GetAllJournalEntries())
+                    {
+                        var usages = entry.GetUsageDetails();
```

Comment:
This needs to be `GetLockDetails()`.

Reply:
Interestingly, the test above succeeds, even though this doesn't actually do what we want at the moment.  Any ideas how we could modify the test to make it fail, before we go ahead and fix this?